src/node/mod.rs

```diff
@@ -0,0 +1,384 @@
+    Control,
+
+    /// A render node may be used by a client to allocate buffers.
+    ///
+    /// Mode-setting is not possible with a render node.
+    Render,
+}
+
+impl NodeType {
+    /// Returns a string representing the prefix of a minor device's name.
+    ///
+    /// For example, on Linux with a primary node, the returned string would be `card`.
+    pub fn minor_name_prefix(&self) -> &'static str {
+        match self {
+            NodeType::Primary => PRIMARY_NAME,
+            NodeType::Control => CONTROL_NAME,
+            NodeType::Render => RENDER_NAME,
+        }
+    }
+
+    #[cfg(any(target_os = "freebsd", target_os = "openbsd"))]
+    fn minor_base(&self) -> u32 {
+        match self {
+            NodeType::Primary => 0,
+            NodeType::Control => 64,
+            NodeType::Render => 128,
+        }
+    }
+}
+
+impl Display for NodeType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+/// An error that may occur when creating a DrmNode from a file descriptor.
+#[derive(Debug)]
+pub enum CreateDrmNodeError {
+    /// Some underlying IO error occured while trying to create a DRM node.
+    Io(io::Error),
+
+    /// The provided file descriptor does not refer to a DRM node.
+    NotDrmNode,
+}
+
+impl Display for CreateDrmNodeError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Io(err) => Display::fmt(err, f),
+            Self::NotDrmNode => {
+                f.write_str("the provided file descriptor does not refer to a DRM node")
+            }
+        }
+    }
+}
+
+impl Error for CreateDrmNodeError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            Self::Io(err) => Some(err),
+            Self::NotDrmNode => None,
+        }
+    }
+}
+
+impl From<io::Error> for CreateDrmNodeError {
+    #[inline]
+    fn from(err: io::Error) -> Self {
+        CreateDrmNodeError::Io(err)
+    }
+}
+
+#[cfg(target_os = "freebsd")]
+fn devname(dev: dev_t) -> Option<String> {
+    use std::os::raw::{c_char, c_int};
+
+    // Matching value of SPECNAMELEN in FreeBSD 13+
+    let mut dev_name = vec![0u8; 255];
+
+    let buf: *mut c_char = unsafe {
+        libc::devname_r(
+            dev,
+            libc::S_IFCHR, // Must be S_IFCHR or S_IFBLK
+            dev_name.as_mut_ptr() as *mut c_char,
+            dev_name.len() as c_int,
+        )
+    };
+
+    // Buffer was too small (weird issue with the size of buffer) or the device could not be named.
+    if buf.is_null() {
+        return None;
+    }
+
+    // SAFETY: The buffer written to by devname_r is guaranteed to be NUL terminated.
+    unsafe { dev_name.set_len(libc::strlen(buf)) };
+
+    Some(String::from_utf8(dev_name).expect("Returned device name is not valid utf8"))
+}
+
+/// Returns if the given device by major:minor pair is a drm device
```

Comment:
Here and below to match the rest of your doc-comments:
```suggestion
/// Returns if the given device by `major:minor` pair is a DRM device.
```

But since this is not an accurate representation (it's just a number), switch that to `the DRM device described by major and minor device numbers` that you're using in other doc-comments?

```suggestion
/// Returns if the given device described by major and minor device numbers is a DRM device.
```

Reply:
I don't think there's anything wrong with the current way.

Reply:
It's just inconsistent with some other functions that take a `dev: dev_t`, that's all.